src/acceleration/IQNILSAcceleration.cpp

```diff
@@ -235,31 +235,35 @@ void IQNILSAcceleration::specializedIterationsConverged(
     const DataMap &cplData)
 {
   PRECICE_TRACE();
-  if (_matrixCols.front() == 0) { // Did only one iteration
-    _matrixCols.pop_front();
-  }
+  if (_matrixCols.size() > 0) {
```

Comment:
We should use a guard here for the sake of readability:

```cpp
if (_matrixCols.empty()) [
 PRECICE_WARN("...");
 return;
}
```

This makes it clear that the rest of the code can assume `_matrixCols.size() > 0`.

Reply:
I moved the warning to the front and used `.empty()` instead of `.size() > 0`. That's indeed clearer.